libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewEventDispatcher.java

```diff
@@ -167,14 +173,22 @@ void onInstructionListVisibilityChanged(boolean shown) {
 
   SpeechAnnouncement onAnnouncement(SpeechAnnouncement announcement) {
     if (speechAnnouncementListener != null) {
-      return speechAnnouncementListener.willVoice(announcement);
+      String textAnnouncement = speechAnnouncementListener.willVoice(announcement);
+      SpeechAnnouncement announcementToBeVoiced = SpeechAnnouncement.builder().announcement(textAnnouncement).build();
+      return announcementToBeVoiced;
     }
     return announcement;
   }
 
   BannerInstructions onBannerDisplay(BannerInstructions instructions) {
     if (bannerInstructionsListener != null) {
-      return bannerInstructionsListener.willDisplay(instructions);
+      String textBanner = bannerInstructionsListener.willDisplay(instructions);
+      List<BannerComponents> bannerComponents = new ArrayList<>(ONE);
```

Comment:
can you rename this `bannerComponentsList`? It's confusing since the object name is plural (`BannerComponents`).

Reply:
Yeah, as you mentioned the problem here is that the object name is plural (not the best one though IMO). Will try to come up with a better name and update. Thanks for flagging.